saleor/graphql/account/mutations.py

```diff
@@ -148,6 +152,32 @@ class Meta:
         model = models.User
 
 
+class LoggedCustomerUpdate(CustomerCreate):
+    class Arguments:
+        id = graphene.ID(
+            description='ID of a customer to update.', required=True)
+        input = UserAddressInput(
+            description='Fields required to update logged in customer.',
+            required=True)
+
+    class Meta:
+        description = 'Updates an existing logged in customer.'
+        exclude = ['password']
+        model = models.User
+
+    @classmethod
+    def user_is_allowed(cls, user, input):
+        id = graphene.Node.to_global_id('User', user.id)
+        return user.has_perm('account.manage_users') or id == input['id']
+
+    @classmethod
+    def mutate(cls, root, info, **data):
```

Comment:
Addresses should be validated in here, there's `i18nMixin` with `validate_address` function

Reply:
Have not seen that this mutation inherits from `CustomerCreate` which already validates the addresses